hummingbot/market/dolomite/dolomite_market

```diff
@@ -0,0 +1,1362 @@
+
+    def to_json(self) -> Dict[str, any]:
+        return {
+            "client_order_id": self.client_order_id,
+            "exchange_order_id": self.exchange_order_id,
+            "symbol": self.symbol,
+            "is_buy": self.is_buy,
+            "order_type": self.order_type.name,
+            "amount": str(self.amount),
+            "available_amount": str(self.available_amount),
+            "price": str(self.price),
+            "executed_amount": str(self.executed_amount),
+            "quote_asset_amount": str(self.quote_asset_amount),
+            "fee_amount": str(self.fee_amount),
+            "last_state": self.last_state,
+        }
+
+    @classmethod
+    def from_json(cls, data: Dict[str, any]) -> "InFlightOrder":
+        cdef:
+            InFlightOrder retval = InFlightOrder(
+                data["client_order_id"],
+                data["exchange_order_id"],
+                data["symbol"],
+                data["is_buy"],
+                getattr(OrderType, data["order_type"]),
+                Decimal(data["amount"]),
+                Decimal(data["price"]),
+            )
+        retval.available_amount = Decimal(data["available_amount"])
+        retval.executed_amount = Decimal(data["executed_amount"])
+        retval.quote_asset_amount = Decimal(data["quote_asset_amount"])
+        retval.fee_amount = Decimal(data["fee_amount"])
+        retval.last_state = data["last_state"]
+        return retval
+
+
+ZERO_EX_MAINNET_PROXY = "0xb258f5C190faDAB30B5fF0D6ab7E32a646A4BaAe" #Loopring protocol delegate address
+
+
+cdef class DolomiteMarket(MarketBase):
+    
+    MARKET_RECEIVED_ASSET_EVENT_TAG = MarketEvent.ReceivedAsset.value
+    MARKET_BUY_ORDER_COMPLETED_EVENT_TAG = MarketEvent.BuyOrderCompleted.value
+    MARKET_SELL_ORDER_COMPLETED_EVENT_TAG = MarketEvent.SellOrderCompleted.value
+    MARKET_WITHDRAW_ASSET_EVENT_TAG = MarketEvent.WithdrawAsset.value
+    MARKET_ORDER_CANCELLED_EVENT_TAG = MarketEvent.OrderCancelled.value
+    MARKET_ORDER_FILLED_EVENT_TAG = MarketEvent.OrderFilled.value
+    MARKET_ORDER_FAILURE_EVENT_TAG = MarketEvent.OrderFailure.value
+    MARKET_BUY_ORDER_CREATED_EVENT_TAG = MarketEvent.BuyOrderCreated.value
+    MARKET_SELL_ORDER_CREATED_EVENT_TAG = MarketEvent.SellOrderCreated.value
+
+    API_CALL_TIMEOUT = 10.0
+    UPDATE_TRADE_FEES_INTERVAL = 60 * 60
+    ORDER_EXPIRY_TIME = 15 * 60.0
+    CANCEL_EXPIRY_TIME = 60.0
+    
+    DOLOMITE_REST_ENDPOINT = "https://exchange-api.dolomite.io"
+    
+    
+    
+    infura_url = "https://mainnet.infura.io/v3/d2de8f9c500e4a5c9761ef1ffdd98cc6"
+    web3 = Web3(Web3.HTTPProvider(infura_url))
+    
+    
+    
+    #Tokens by contract addresses 
+    
+    tickers = {
+        
+        'WETH': '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
+        'DAI': '0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359',
+        'LRC': '0xBBbbCA6A901c926F240b89EacB641d8Aec7AEafD',
+        'BAT': '0x0D8775F648430679A709E98d2b0Cb6250d2887EF'
+        
+    }
```

Comment:
Why are these hardcoded?

Reply:
I made the dictionary above to represent the tokens that the currently supported markets on Dolomite offers and their corresponding contract addresses. I'm using this in the prepare order route within the "market" and "fee_token_address" fields.

Reply:
Just fixed this - created a method that uses the tokens route to fetch the addresses now so no more hardcoding!